### ISSUE TITLE
Initialization lists love you... Love them back :)

### ISSUE THREAD

**Issue by Elried** [8INF840-uqac/Devoir1#15]
In Node.inl we have the following code :
```cpp
template <class T>
Node<T>::Node(const T id,
std::shared_ptr<Node<T>> previous_node,
std::shared_ptr<Node<T>> next_node,
std::shared_ptr<std::mutex> list_mutex) {
    m_sum = std::make_shared<T>(T());
    m_state_sum = std::make_shared<short>(0);
    m_destroyed = false;
    m_id = std::make_shared<T>(id);
    m_previous_node = previous_node;
    m_next_node = next_node;
    m_sum_thread = std::thread([this, list_mutex]() {
        while (!this->m_destroyed) {
            // We need to lock the list to get the good sum and avoid removed items
            if (list_mutex->try_lock()) {
                T result = T();
                // update sum
                this->send(result);
                // We can read the correct sum
                std::cout << "Sum - node " << *m_id <<  ":  ";
                std::cout << *this->m_sum << std::endl;
                list_mutex->unlock();
            }
        }
    });
}
```

The issue here is that initialization lists are not used. That creates runtime overhead as [this snippet](http://coliru.stacked-crooked.com/a/2553ed1498fa0684) demonstrates.

TLDR : Use initialization list :)
